File: javascript/resistor-color/resistor-color.js
export const colorCode = (colorString) => {

  let color = colorString.toLowerCase()

  for (let i = 0; i < COLORS.length; i++) {
    if (COLORS[i] === color) {
      return i
    }
  }
}

export const COLORS = ["black","brown","red","orange","yellow","green","blue","violet","grey","white"];

File: csharp/resistor-color/ResistorColor.cs
using System;

public static class ResistorColor
{
    private static string[] colours = { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };
    public static int ColorCode(string color)
    {
        for (int i = 0; i < colours.Length; i++)
        {
            if (color == colours[i])
            {
                return i;
            }
        }
        return 0;
    }

    public static string[] Colors()
    {
        return colours;
    }
}
